bee-crypto/src/ternary/bigint/i384/mod.rs

```diff
@@ -0,0 +1,893 @@
+        if numbers_negative {
+            for (s, o) in zipped_iter {
+                if s > o {
+                    return Some(Less);
+                } else if s < o {
+                    return Some(Greater);
+                }
+            }
+        } else {
+            for (s, o) in zipped_iter {
+                if s > o {
+                    return Some(Greater);
+                } else if s < o {
+                    return Some(Less);
+                }
+            }
+        }
+
+        Some(Equal)
+    }
+}
+
+impl Ord for I384<BigEndian, U8Repr> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.partial_cmp(other) {
+            Some(ordering) => ordering,
+
+            // The ordering is total, hence `partial_cmp` will never return `None`.
+            None => unreachable!(),
+        }
+    }
+}
+
+impl From<T242<Btrit>> for I384<BigEndian, U8Repr> {
+    fn from(value: T242<Btrit>) -> Self {
+        let i384_le: I384<LittleEndian, U32Repr> = value.into();
+        let i384_be: I384<BigEndian, U32Repr> = i384_le.into();
+        i384_be.into()
+    }
+}
+
+impl TryFrom<T243<Btrit>> for I384<BigEndian, U8Repr> {
+    type Error = Error;
+
+    fn try_from(value: T243<Btrit>) -> Result<Self, Error> {
+        let i384_le: I384<LittleEndian, U32Repr> = value.try_into()?;
+        let i384_be: I384<BigEndian, U32Repr> = i384_le.into();
+        Ok(i384_be.into())
+    }
+}
+
+impl I384<BigEndian, U32Repr> {
+    /// Adds `other` onto `self` in place.
+    pub fn add_inplace(&mut self, other: Self) {
+        let mut overflown = false;
+        let self_iter = self.inner.iter_mut().rev();
+        let other_iter = other.inner.iter().rev();
+
+        for (s, o) in self_iter.zip(other_iter) {
+            let (sum, still_overflown) = s.overflowing_add_with_carry(*o, overflown as u32);
+            *s = sum;
+            overflown = still_overflown;
+        }
+    }
+
+    /// Adds `other` in place, returning the number of digits required accomodate `other` (starting from the least
+    /// significant one).
+    pub fn add_digit_inplace<T: Into<u32>>(&mut self, other: T) -> usize {
+        let other = other.into();
+
+        let mut i = self.inner.len() - 1;
+
+        let (sum, mut overflown) = self.inner[i].overflowing_add(other);
+        self.inner[i] = sum;
+
+        i -= 1;
+
+        while overflown {
+            let (sum, still_overflown) = self.inner[i].overflowing_add(1u32);
+            self.inner[i] = sum;
+            overflown = still_overflown;
+            i -= 1;
+        }
+
+        i
+    }
+
+    /// Reinterprets the I384 as an U384.
+    pub fn as_u384(self) -> U384<BigEndian, U32Repr> {
+        U384::<BigEndian, U32Repr>::from_array(self.inner)
+    }
+
+    /// Creates an I384 from a balanced T242.
+    pub fn from_t242(value: T242<Btrit>) -> Self {
+        // First make it unbalanced.
+        let t242_unbalanced = value.into_shifted();
```

Comment:
I don't find `into_shifted` very self-describing, which also seemed to make the comment here necessary. Is it possible to rename that method into `into_unbalanced` or even `into_unbalanced_ternary`?